cypress/e2e/devpool.cy.ts

```diff
@@ -0,0 +1,146 @@
+    cy.get("#issues-container > :nth-child(1)").should("not.have.class", "new-task");
+
+    // needed to make sure data is written to the local storage
+    cy.wait(3000);
+
+    cy.log("Should display two new tasks");
+    cy.clock(Date.now() + 95000000);
+    cy.visit("/");
+    const fakeNow = new Date("2022-04-10");
+    // Needed due to a bug
+    cy.clock(fakeNow).then((clock) => {
+      // @ts-expect-error https://github.com/cypress-io/cypress/issues/7577
+      return clock.bind(window);
+    });
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+  });
+
+  it("Display a message on rate limited", () => {
+    cy.intercept("https://api.github.com/repos/*/*/issues**", (req) => {
+      req.reply({
+        statusCode: 403,
+      });
+    }).as("getIssues");
+    cy.visit("/");
+    cy.get(".preview-header").should("exist");
+  });
+
+  it("Items can be sorted", () => {
+    cy.intercept("https://api.github.com/repos/*/*/issues**", (req) => {
+      req.reply({
+        statusCode: 200,
+        body: [issue1, issue2],
+      });
+    }).as("getIssues");
+    cy.visit("/");
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="price"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="price"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="time"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="time"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="priority"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="priority"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="activity"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get('[for="activity"]').click();
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+    cy.get("#filter").type("draft");
+    cy.get('div[id="issues-container"]').children().should("have.length", 2);
+  });
+
+  it("User can log in", () => {
+    cy.intercept("https://api.github.com/repos/*/*/issues**", (req) => {
+      req.reply({
+        statusCode: 200,
+        body: [issue1, issue2],
+      });
+    }).as("getIssues");
+    cy.intercept("https://api.github.com/user", (req) => {
+      req.reply({
+        statusCode: 200,
+        body: githubUser,
+      });
+    }).as("getUser");
+    cy.intercept("https://iyybhhiflwbsjopsgaow.supabase.co/auth/v1/authorize?provider=github", (req) => {
+      req.reply({
+        statusCode: 200,
+      });
+      // Simulate login token
+      window.localStorage.setItem("sb-wfzpewmlyiozupulbuur-auth-token", JSON.stringify(loginToken));
```

Comment:
I think real end-to-end testing seems like a missed opportunity here. 

If it's for speed reasons I think it would be cool to have both versions (mocks and real)

Could be interesting depending on how long it takes to run it on lint-staged for any UI code. And the real runs can happen when we push to GitHub

Reply:
Same comment as above, it's like testing some code that isn't ours. We could test the speed indeed but it's biased since there is not UI running technically. We could have real e2e testing mixed there, though. The advantage of mocking it is that it is also very easy to setup locally or on a testing machine because no real tokens are needed, and no bandwidth is used.
You can have a read about pros and cons in the interesting piece of documentation
https://docs.cypress.io/guides/guides/network-requests#Use-Server-Responses